modules/spring-config/src/main/java/org/mule/config/spring/MuleArtifactContext.java

```diff
@@ -118,10 +184,56 @@ protected BeanDefinitionReader createBeanDefinitionReader(DefaultListableBeanFac
         beanDefinitionReader.setDocumentReaderClass(getBeanDefinitionDocumentReaderClass());
         //add error reporting
         beanDefinitionReader.setProblemReporter(new MissingParserProblemReporter());
+        registerAnnotationConfigProcessors(beanDefinitionReader.getRegistry(), null);
 
         return beanDefinitionReader;
     }
 
+    private void registerAnnotationConfigProcessors(BeanDefinitionRegistry registry, Object source)
+    {
+
+        if (!registry.containsBeanDefinition(CONFIGURATION_ANNOTATION_PROCESSOR_BEAN_NAME))
+        {
+            RootBeanDefinition def = new RootBeanDefinition(ConfigurationClassPostProcessor.class);
+            def.setSource(source);
+            registerPostProcessor(registry, def, CONFIGURATION_ANNOTATION_PROCESSOR_BEAN_NAME);
+        }
+
+        if (!registry.containsBeanDefinition(AUTOWIRED_ANNOTATION_PROCESSOR_BEAN_NAME))
+        {
+            RootBeanDefinition def = new RootBeanDefinition(AutowiredAnnotationBeanPostProcessor.class);
+            def.setSource(source);
+            registerPostProcessor(registry, def, AUTOWIRED_ANNOTATION_PROCESSOR_BEAN_NAME);
+        }
+
+        if (!registry.containsBeanDefinition(AnnotationConfigUtils.REQUIRED_ANNOTATION_PROCESSOR_BEAN_NAME))
+        {
+            RootBeanDefinition def = new RootBeanDefinition(RequiredAnnotationBeanPostProcessor.class);
+            def.setSource(source);
+            registerPostProcessor(registry, def, AnnotationConfigUtils.REQUIRED_ANNOTATION_PROCESSOR_BEAN_NAME);
+        }
+
+        registerJsr250PostProcessors(registry, source);
+    }
+
+    private void registerJsr250PostProcessors(BeanDefinitionRegistry registry, Object source)
+    {
+        RootBeanDefinition def = new RootBeanDefinition(FilteringCommonAnnotationBeanPostProcessor.class);
+        ConstructorArgumentValues arguments = new ConstructorArgumentValues();
+        Set<String> filteredPackages = new HashSet<>();
+        filteredPackages.add("org.apache.cxf");
+        arguments.addGenericArgumentValue(filteredPackages);
+        def.setConstructorArgumentValues(arguments);
+        def.setSource(source);
+        registerPostProcessor(registry, def, COMMON_ANNOTATION_PROCESSOR_BEAN_NAME);
+    }
+
+    private void registerPostProcessor(BeanDefinitionRegistry registry, RootBeanDefinition definition, String beanName)
+    {
+        definition.setRole(BeanDefinition.ROLE_INFRASTRUCTURE);
+        registry.registerBeanDefinition(beanName, definition);
```

Comment:
yes, because the class been registered is ours not spring. Actually, the other checks are not really necessary. will remove them